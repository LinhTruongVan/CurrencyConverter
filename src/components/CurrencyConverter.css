.currency__input-group{
    margin-top: 112px;
}

.currency__title,
.currency__converting-result {
    font-size: 72px;
    font-weight: bold;
    font-style: normal;
    font-stretch: normal;
    line-height: normal;
    letter-spacing: normal;
    color: #c05926;
    text-align: center;
}

.currency__amount-input{
    width: 400px;
    height: 90px;
    background-color: #ffffff;
    border: solid 1px #268dc0;

    font-size: 36px;
    font-weight: normal;
    font-style: normal;
    font-stretch: normal;
    line-height: normal;
    letter-spacing: normal;
    text-align: right;
    color: #615e5f;

}

.currency__selector{
    width: 400px;
    height: 90px;
    background-color: #59b5db;
    border: solid 1px #268dc0;

    font-size: 36px;
    font-weight: bold;
    font-style: normal;
    font-stretch: normal;
    line-height: normal;
    letter-spacing: normal;
    text-align: left;
    color: #ffffff;

    -webkit-appearance: none;
    -webkit-border-radius: 0px;
}

.currency__selector-caret{
    font-size: 36px;
    font-weight: normal;
    font-style: normal;
    font-stretch: normal;
    line-height: normal;
    letter-spacing: normal;
    text-align: right;
    color: #ffffff;
}

.currency__convert-btn{
    width: 248px;
    height: 94px;
    border-radius: 16px;
    background-color: #c05926;

    font-size: 36px;
    font-weight: bold;
    font-style: normal;
    font-stretch: normal;
    line-height: normal;
    letter-spacing: normal;
    color: #ffffff;

    margin-top: 64px;
}

.currency__convert-btn:focus,
.currency__convert-btn:hover{
    color: #ffffff;
}