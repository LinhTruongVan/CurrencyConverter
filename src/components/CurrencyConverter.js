// @flow
import React, { Component } from 'react';

import './CurrencyConverter.css';

type Props = {
  currencies: { [key: string]: number },
  test: boolean
};

type State = {
  amount: number,
  selectedCurrency: string
}

class CurrencyConverter extends Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = { amount: 0, selectedCurrency: Object.keys(this.props.currencies)[0] };
    (this: any).onAmountChange = this.onAmountChange.bind(this);
    (this: any).onCurrencyChange = this.onCurrencyChange.bind(this);
  }

  onAmountChange(e: SyntheticInputEvent<HTMLInputElement>) {
    let amount = parseFloat(e.target.value);
    if (isNaN(amount)) amount = 0;
    this.setState({ amount: amount });
  }

  convertCurrency() {
    return (this.state.amount * this.props.currencies[this.state.selectedCurrency]).toFixed(2);
  }

  renderCurrenySelector() {
    return (
      <select className="currency__selector" onChange={this.onCurrencyChange}>
        {
          Object.keys(this.props.currencies)
            .map(currency => <option key={currency} value={currency}>{currency}</option>)
        }
      </select>
    );
  }

  onCurrencyChange(e: SyntheticInputEvent<HTMLSelectElement>) {
    this.setState({ selectedCurrency: e.target.value });
  }

  render() {
    return (
      <div>
        <div className="currency__title">{this.state.selectedCurrency}</div>
        <div className="currency__converting-result">{this.convertCurrency()}</div>

        <div className="input-group currency__input-group">
          <input type="number" className="currency__amount-input" value={this.state.amount} onChange={this.onAmountChange} />
          {this.renderCurrenySelector()}
        </div>

      </div>
    );
  }
}

export default CurrencyConverter;
