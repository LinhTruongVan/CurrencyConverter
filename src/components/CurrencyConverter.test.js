import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';

import CurrencyConverter from './CurrencyConverter';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
    const props = {
        currencies: {
            USD: 1,
            EUR: 2,
            JPY: 3
        }
    };
    console.log(1);
    const enzymeWrapper = mount(<CurrencyConverter {...props} />);
    return {
        props,
        enzymeWrapper
    }
}

describe('components', () => {
    describe('CurrencyConverter', () => {
        it('should render self', () => {
            const { enzymeWrapper, props } = setup();

            expect(enzymeWrapper.find('input').exists()).toBeTruthy()
            expect(enzymeWrapper.find('select').exists()).toBeTruthy()
            expect(enzymeWrapper.find('select option').length).toBe(Object.keys(props.currencies).length)
        })

        it('should update amount after user typing amount', () => {
            const { enzymeWrapper } = setup();

            const expectedValue = '7';
            const input = enzymeWrapper.find('input').first();
            input.simulate('change', { target: { value: expectedValue } });

            expect(input.instance().value).toEqual(expectedValue);
        })
    })
})