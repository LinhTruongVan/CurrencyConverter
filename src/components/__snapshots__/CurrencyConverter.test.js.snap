// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CurrencyConverter component correctly 1`] = `
<div>
  <div
    className="currency__title"
  >
    USD
  </div>
  <div
    className="currency__result"
  >
    0.00
  </div>
  <div
    className="input-group currency__input-group"
  >
    <input
      className="currency__amount-input"
      onChange={[Function]}
      type="number"
      value={0}
    />
    <select
      className="currency__selector"
      onChange={[Function]}
    >
      <option
        value="USD"
      >
        USD
      </option>
      <option
        value="EUR"
      >
        EUR
      </option>
      <option
        value="JPY"
      >
        JPY
      </option>
    </select>
  </div>
</div>
`;

exports[`renders component 1`] = `
<div>
  <div
    className="currency__title"
  >
    USD
  </div>
  <div
    className="currency__result"
  >
    0.00
  </div>
  <div
    className="input-group currency__input-group"
  >
    <input
      className="currency__amount-input"
      onChange={[Function]}
      type="number"
      value={0}
    />
    <select
      className="currency__selector"
      onChange={[Function]}
    >
      <option
        value="USD"
      >
        USD
      </option>
      <option
        value="EUR"
      >
        EUR
      </option>
      <option
        value="JPY"
      >
        JPY
      </option>
    </select>
  </div>
</div>
`;
